src/rust/src/lazy/dataframe.rs

```diff
@@ -80,11 +77,77 @@ impl LazyFrame {
         })
     }
 
-    pub fn collect_handled(&self) -> crate::rpolarserr::RResult<crate::rdataframe::DataFrame> {
+    pub fn collect_handled(&self) -> RResult<crate::rdataframe::DataFrame> {
         use crate::rpolarserr::WithRctx;
         handle_thread_r_requests(self.clone().0).when("calling $collect() on LazyFrame")
     }
 
+    pub fn sink_parquet(
+        &self,
+        path: Robj,
+        compression_method: Robj,
+        compression_level: Robj,
+        statistics: Robj,
+        row_group_size: Robj,
+        data_pagesize_limit: Robj,
+        maintain_order: Robj,
+    ) -> RResult<()> {
+        use polars::prelude::ParquetCompression::*;
+        let pqcomp = match robj_to!(String, compression_method)?.as_str() {
+            "uncompressed" => Ok(Uncompressed),
+            "snappy" => Ok(Snappy),
+            "gzip" => robj_to!(Option, u8, compression_level)?
+                .map(polars::prelude::GzipLevel::try_new)
+                .transpose()
+                .map(Gzip),
+            "lzo" => Ok(Lzo),
+            "brotli" => robj_to!(Option, u32, compression_level)?
+                .map(polars::prelude::BrotliLevel::try_new)
+                .transpose()
+                .map(Brotli),
+            "zstd" => robj_to!(Option, i64, compression_level)?
```

Comment:
I believe we cannot directly use `robj_to!(Option, i32, ...)` because we do not have `robj_to_i32`. We only have `robj_to_i64`.

Reply:
oh yes my bad. But we should have i32 and f64. I will impl those conversions now.


I'm a bit pedantic about it, because it is good practice in rust to only use `as` if the new type do fully contain the old type.  So `i32 as i64` is always fine. `u32 as i32` is not. 

rust will silently wrap overflows
```rust
fn main() {
    println!("max {} min {}",i64::MAX as i32, i64::MIN as i32);
}
```
output is
```
max -1 min 0
```

Reply:
Undeniably this is a good practice, but since the zstd compression levels should range from -7 to 22, I thought a `i64` should be no different from a `i32` here (still we should use `robj_to_i32` if possible). We should probably check this on the R side.

Reply:
yeah it is very exotic, sry ^^

I have added `robj_to!(f64/i32, )` to this PR now

Reply:
oups added it to #323  :/ , well needed to merge in CSE anyways